openpype/modules/sync_server/sync_server.py

```diff
@@ -229,6 +244,220 @@ def _get_configured_sites_from_setting(module, project_name, project_setting):
+    representation_ids.update(
+        get_linked_representation_id(
+            project_name, repre_id=workfile_representation["_id"]
+        )
+    )
+    for repre_id in representation_ids:
+        sync_server.add_site(
+            project_name,
+            repre_id,
+            local_site_id,
+            force=True,
+            priority=99,
+            reset_timer=True,
+        )
+
+    # While representation unavailable locally, wait.
+    while not sync_server.is_representation_on_site(
+        project_name, workfile_representation["_id"], local_site_id
+    ):
+        sleep(5)
+
+    if not last_published_workfile_path:
+        last_published_workfile_path = get_last_published_workfile_path(
+            host_name,
+            project_name,
+            task_name,
+            workfile_representation,
+            anatomy=anatomy,
+        )
+
+    # Get workfile data
+    workfile_data = get_template_data_with_names(
+        project_name, asset_name, task_name, host_name
+    )
+
+    extension = last_published_workfile_path.split(".")[-1]
+
+    project_settings = get_project_settings(project_name)
+    template_key = get_workfile_template_key(
+        task_name, host_name, project_name, project_settings
+    )
+
+    # Get version patter for regex search
+    version_pattern = anatomy.templates[template_key]["version"]
+    version_pattern = re.sub(
+        r"{version.*?}",
+        r"([0-9]+)",
+        version_pattern,
+    )
+
+    # Get local workfile version number
+    last_local_workfile_version = None
+    for filename in sorted(
+        os.listdir(
+            get_workdir(
+                get_project(project_name),
+                asset_doc,
+                task_name,
+                host_name,
+                anatomy=anatomy,
+                template_key=template_key,
+                project_settings=project_settings,
+            )
+        ),
+        reverse=True,
+    ):
+        if filename.endswith(extension):
+            match = re.search(
+                version_pattern,
+                filename
+            )
+            if match:
+                last_local_workfile_version = int(match.group(1))
+                break
+
+    # Set workfile data workfile version
+    # Either last published version or last local version, whichever is higher
+    workfile_data["version"] = (
+        last_local_workfile_version + 1
+        if (
+            last_local_workfile_version
+            and last_local_workfile_version > last_version_doc["name"]
+        )
+        else last_version_doc["name"] + 1
+    )
+    workfile_data["ext"] = extension
+
+    anatomy_result = anatomy.format(workfile_data)
+    local_workfile_path = anatomy_result[template_key]["path"]
+
+    # Copy last published workfile to local workfile directory
+    shutil.copy(
+        last_published_workfile_path,
+        local_workfile_path,
+    )
+
+    return local_workfile_path
```

Comment:
In fact, `download_last_published_workfile` shouldn't return the local path but the last published wf which has been downloaded. The resolving of the `local_workfile_path`  must be put into another function and the copy made my the master process or into a utility function.